receiver/awss3receiver/s3intf.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awss3receiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver"
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+var downloadManager *manager.Downloader //nolint:golint,unused
+
+type ListObjectsV2Pager interface {
+	HasMorePages() bool
+	NextPage(context.Context, ...func(*s3.Options)) (*s3.ListObjectsV2Output, error)
+}
+
+type ListObjectsAPI interface {
+	NewListObjectsV2Paginator(params *s3.ListObjectsV2Input) ListObjectsV2Pager
+}
+
+type GetObjectAPI interface {
+	GetObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+}
+
+type s3ListObjectsAPIImpl struct {
+	client *s3.Client
+}
+
+func newS3Client(cfg S3DownloaderConfig) (ListObjectsAPI, GetObjectAPI, error) {
+	optionsFuncs := make([]func(*config.LoadOptions) error, 0)
+	if cfg.Region != "" {
+		optionsFuncs = append(optionsFuncs, config.WithRegion(cfg.Region))
+	}
+
+	if cfg.Endpoint != "" {
+		customResolver := aws.EndpointResolverWithOptionsFunc(func(_, _ string, _ ...any) (aws.Endpoint, error) {
+			return aws.Endpoint{
+				PartitionID:   "aws",
```

Comment:
Should the AWS partition be user-configurable? Will "aws" always work for configuration, or at least work for the large majority of users?

Reply:
This one is a bit of a magic value taken from localstacks configuration of the aws sdk go v2: https://docs.localstack.cloud/user-guide/integrations/sdks/go/

Reply:
Do you think there would ever be value in allowing the user to configure this? I'm not very familiar with AWS and how it's using partition ID here.

It doesn't have to be done in this PR if it is valuable, just want to get a better idea of what's going on here.

Reply:
It's always possible - after all why make it configurable at the AWS SDK level if it's only ever going to equal "aws"!
After a bit more digging on this I found the partition IDs listed in the v1 sdk: https://github.com/aws/aws-sdk-go/blob/8888dae11935b652781010304c637406a168475a/aws/endpoints/defaults.go#L11
It won't take much to add the configuration option - I'll do it now.